src/OrchardCore/OrchardCore.ContentManagement.Display/ContentItemDisplayManager.cs

```diff
@@ -80,7 +87,27 @@ public async Task<IShape> BuildDisplayAsync(ContentItem contentItem, IUpdateMode
             var metadata = itemShape.Metadata;
             metadata.DisplayType = actualDisplayType;
 
-            // [Stereotype]_[DisplayType]__[ContentType] e.g. Content-BlogPost.Summary
+            if (hasStereotype)
+            {
+                if (actualDisplayType != "Detail")
+                {
+                    // Add fallback/default alternate Stereotype_[DisplayType] e.g. Content.Summary
+                    metadata.Alternates.Add($"Stereotype_{actualDisplayType}");
+
+                    // [Stereotype]_[DisplayType] e.g. Menu.Summary
+                    metadata.Alternates.Add($"{stereotype}_{actualDisplayType}");
+                }
+                else
+                {
+                    // Add fallback/default alternate i.e. Content 
+                    metadata.Alternates.Add("Stereotype");
+
+                    // Add alternate to make the type [Stereotype] e.g. Menu
+                    metadata.Alternates.Add(stereotype);
```

Comment:
here `stereotype` == actualShapeType  you can keep outside if

                    `metadata.Alternates.Add(actualShapeType  );`

Reply:
`metadata.Alternates.Add(actualShapeType  )` is only applicable when `hasStereotype is true` and `actualDisplayType is empty or equals to "Detail"`. Similarly here, the order of the insertion is important here to allow `{stereotype}` to take presence when it exists otherwise we fallback on `stereotype`

Reply:
@ns8482e are you good with my last comment, or see a reason for change?

Reply:
LGTM